src/tty/unix.rs

```diff
@@ -52,7 +52,23 @@ fn get_win_size<T: AsRawFd + ?Sized>(fileno: &T) -> (usize, usize) {
     unsafe {
         let mut size: libc::winsize = zeroed();
         match win_size(fileno.as_raw_fd(), &mut size) {
-            Ok(0) => (size.ws_col as usize, size.ws_row as usize), // TODO getCursorPosition
+            Ok(0) => {
+                // In linux pseudo-terminals are created with dimensions of
+                // zero. If host application didn't initialize the correct
+                // size before start we treat zero size as 80 columns and
+                // inifinite rows
+                let cols = if size.ws_col == 0 {
+                    80
+                } else {
+                    size.ws_col as usize
+                };
+                let rows = if size.ws_row == 0 {
+                    0
+                } else {
+                    usize::max_value()
+                };
```

Comment:
```suggestion
                let rows = if size.ws_row == 0 {
                    usize::max_value()
                } else {
                    size.ws_row as usize
                };
```

Reply:
Also should it use 24 as fallback instead of `usize::max_value()`?

Reply:
This is a good question actually, my point is:
1. There are two cases where rows are used: when scrolling something (e.g. completions), and for vertical scroll once https://github.com/kkawakam/rustyline/pull/378 is landed
2. There is one primary case when terminal size is zero: when using some automation, like rexpect so there is no

So having no vertical scroll for automation is good idea.

In case there is a bug, like this one: https://github.com/containers/libpod/issues/351 This kind of degradation is good: if we have 24 rows and input is large enough that doesn't fit 24 rows, it maybe unclear for user why part of their input is hidden. And if their terminals is smaller than 24 rows (which may easily happen for split windows) no vertical scroll looks nicer than vertical scroll that is larger than an actual terminal.